### PR TITLE
FW-4129 Image thumbnail generation

### PR DESCRIPTION
**Pull request by NolanVH**
### Description of Changes
These changes add three fields to the Image model:
- `thumbnail` (containing a generated image with a max size of 100px)
- `small` (containing a generated image with a max size of 560px)
- `medium` (containing a generated image with a max size of 1000px)

The resized thumbnail images are generated and uploaded to AWS when a new image model is created. When the content field is updated (for example a new image file is uploaded) then the existing images are removed and deleted from AWS and new ones are generated.

The image max sizes for each of the three fields are set in [settings.py](https://github.com/First-Peoples-Cultural-Council/fv-be/blob/8bf767e43372ece8ba0c7aa4e257297577e434b8/firstvoices/firstvoices/settings.py#L268). Resizing works as follows:

- The larger of the input image width or height is scaled down to the size setting and the other dimension is scaled down, keeping the original aspect ratio.
- The original size is used if the input image is smaller in both dimensions than the setting.

The AWS environment variables need to be set as detailed in the README:
- `AWS_ACCESS_KEY_ID`
- `AWS_SECRET_ACCESS_KEY`
- `MEDIA_UPLOAD_S3_BUCKET`
- `MEDIA_UPLOAD_S3_REGION`

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- [x] Admin Panel has been updated if models have been added or modified
- [x] Migrations have been updated and committed if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [x] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

**Review by sarahfirstvoices — APPROVED** [review]
Looking good! My comments are mostly cosmetic

**Review by bronson-c — APPROVED** [review]
LGTM 👍  Tested locally the thumbnails are generated and updated as expected

File: firstvoices/backend/models/media.py [hunk 1]
```diff
@@ -135,9 +141,107 @@ class Meta:
...
+
+    def delete(self, using=None, keep_parents=False):
+        """
+        Deletes the associated media files when the instance is deleted, to prevent orphans.
+        """
+        result = super().delete(using, keep_parents)
+        try:
+            self.thumbnail.delete(save=False)
+            self.small.delete(save=False)
+            self.medium.delete(save=False)
```
[the remark this answers is not in the document]
Comment (NolanVH) [reply firstvoices/backend/models/media.py:236]: The content field is already deleted in the [MediaBase parent class](https://github.com/First-Peoples-Cultural-Council/fv-be/blob/8bf767e43372ece8ba0c7aa4e257297577e434b8/firstvoices/backend/models/media.py#L42-L78). I could add a check in the parent delete function to check if the model type is an Image and delete the thumbnails there if needed.

File: firstvoices/backend/models/media.py [hunk 2]
```diff
@@ -135,9 +141,107 @@ class Meta:
...
+        A function to generate a set of resized images when an Image model is saved
+        """
+        for size_name, max_size in settings.CURRENT_MAX_IMAGE_SIZES.items():
+            output_img = BytesIO()
+
+            img = PILImage.open(self.content)
+            image_name = self.content.name.split(".")[0]
+
+            if img.width >= img.height:
+                # If the input image is larger than the max size, scale down the output image.
```
[the remark this answers is not in the document]
Comment (NolanVH) [reply firstvoices/backend/models/media.py]: I have moved the resize calculation into a helper function.

File: firstvoices/firstvoices/settings.py [hunk 3]
```diff
@@ -263,3 +263,6 @@
 AWS_QUERYSTRING_EXPIRE = (
     60 * 60
 )  # seconds until a query string expires; this is the default setting
+
+# Image thumbnail generation settings
+CURRENT_MAX_IMAGE_SIZES = {"thumbnail": 100, "small": 560, "medium": 1000}
```
[the remark this answers is not in the document]
Comment (NolanVH) [reply firstvoices/firstvoices/settings.py]: The name has been updated. Thanks!

File: firstvoices/backend/models/media.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sarahfirstvoices) [reply firstvoices/backend/models/media.py:245]: Oh, right. Looks good like this! Thanks for pointing this out :)

**Pull request merged**